app/helpers/address_helper.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+module AddressHelper
+  def postcode_lookup(postcode)
+    postcode_url = "#{ENV['API_URL']}?postcode=#{postcode}&key=#{ENV['API_KEY']}"
+
+    response = Typhoeus.get(postcode_url)
+
+    results = []
+    if response.code == 200
+      addresses = JSON.parse(response.body)["results"]
+      addresses.each do |address|
+        results << { address.dig("DPA", "UPRN") => address.dig("DPA", "ADDRESS") }
+      end
+    else
+      results << { response.code => I18n.t("address_api.error_#{response.code}") }
+    end
+
+    { "code" => response.code, "results" => results }.to_json
```

Comment:
I think returning a Rails hash here would be acceptable, since the helper will only be used within this application, so we'd always have to parse the JSON in every use.

Reply:
Thanks @brucebolt - I had initially thought that I'd be implementing all the way onto the UI (now a separate card) - hence the JSON response.  But, absolutely right that is not needed as part of the this card.  Changed as per your suggestion.